validation/src/verifier/event_verifier.rs

```diff
@@ -0,0 +1,61 @@
+use anyhow::Result;
+use ceramic_event::unvalidated::signed;
+use ipld_core::ipld::Ipld;
+
+use super::{cacao_verifier::Verifier as _, key_verifier::Verifier as _, opts::VerifyOpts};
+
+#[async_trait::async_trait]
+pub trait Verifier {
+    async fn verify_signature(&self, opts: &VerifyOpts) -> Result<()>;
```

Comment:
I'm pretty sure this is going to need more parameters or to return something so we can make sure to verify controller and more things but I'm planning to adjust that when I start using it in the next PR.